### PR TITLE
ref(otel): Use new scopes API

### PR DESCRIPTION
**Pull request by sentrivana**
Speedrunning through the remaining integrations 🏃🏻‍♀️

Relates https://github.com/getsentry/sentry-python/issues/2817

---

## General Notes

Thank you for contributing to `sentry-python`!

Please add tests to validate your changes, and lint your code using `tox -e linters`.

Running the test suite on your PR might require maintainer approval. Some tests (AWS Lambda) additionally require a maintainer to add a special label to run and will fail if the label is not present.

#### For maintainers

Sensitive test suites require maintainer review to ensure that tests do not compromise our secrets. This review must be repeated after any code revisions.

Before running sensitive test suites, please carefully check the PR. Then, apply the `Trigger: tests using secrets` label. The label will be removed after any code changes to enforce our policy requiring maintainers to review all code revisions before running sensitive tests.

### CODE REVIEW

File: sentry_sdk/integrations/opentelemetry/span_processor.py [hunk 1]
```diff
@@ -116,25 +114,23 @@ def _prune_old_spans(self):
 
     def on_start(self, otel_span, parent_context=None):
         # type: (OTelSpan, Optional[SpanContext]) -> None
-        hub = Hub.current
-        if not hub:
```
Comment (szokeasaurusrex) [inline sentry_sdk/integrations/opentelemetry/span_processor.py:120]: Does the `dsn` check make this check redundant? Trying to understand why we are removing this

**Review by szokeasaurusrex — COMMENTED** [review]
Looks good mostly, please see the question I left

File: sentry_sdk/integrations/opentelemetry/span_processor.py (hunk 1 above)
Comment (sentrivana) [reply sentry_sdk/integrations/opentelemetry/span_processor.py:120]: What would be the post-hub world equivalent of this check? It seems to me like the purpose of the hub in this function was basically two things: accessing the client and making sure the client has a DSN, and using the hub directly when starting the Sentry span (which we don't want to do anymore in favor of using the top-level API).

Comment (antonpirker) [reply sentry_sdk/integrations/opentelemetry/span_processor.py:120]: As far as I remember those checks where here to just make sure we have a Sentry setup with and DSN where we can send data to.

So I think it is OK to remove the hub check and just check for a valid DSN further down.

**Review by antonpirker — APPROVED** [review]
Looks good!

**Review by szokeasaurusrex — APPROVED** [review]
I also approve, thank you both for addressing my question

**Pull request merged**